Source/Dafny/Compiler-go.cs

```diff
@@ -1831,9 +1831,27 @@ private class SpecialNativeType : UserDefinedType {
     }
 
     protected override void EmitPrintStmt(TargetWriter wr, Expression arg) {
-      wr.Write("_dafny.Print(");
-      TrExpr(arg, wr, false);
-      wr.WriteLine(")");
+      bool isString = arg.Type.AsCollectionType != null &&
+                      arg.Type.AsCollectionType.AsSeqType != null &&
+                      arg.Type.AsCollectionType.AsSeqType.Arg.IsCharType;
+      if (isString) {
+        // needed for empty sequences known to be strings
+        if (arg.Resolved is MemberSelectExpr &&
+            (arg.Resolved as MemberSelectExpr).Member.IsExtern(out _, out _)) {
```

Comment:
As a general rule, I prefer `((MemberSelectExpr)arg.Resolved)` over `arg.Resolved as MemberSelectExpr` when you expect the cast to succeed. It's a way in the source to indicate your understanding that the cast will succeed.

In this case, you can simplify the guard further as:

    arg.Resolved is MemberSelectExpr mse && mse.Member.IsExtern(out _, out _)

Reply:
OK